backend/src/core/middlewares/campaign.middleware.ts

```diff
@@ -151,9 +152,28 @@ const isCampaignRedacted = async (
   }
 }
 
+/**
+ * Validate url origin is from Vault and check url expiry
+ */
+const isValidVaultUrl = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Response | void => {
+  const { url } = req.body
+  const vaultUrl = new URL(url)
+  if (vaultUrl.hostname !== VAULT_DOMAIN) {
+    // OR url has expired, check if url has expired using X-Amz-Expires header
+    return res.status(400).json({ message: 'This is not a valid Vault url' })
+  } else {
+    next()
+  }
```

Comment:
If `url` is not a parsable URL, an error will be thrown and the user will see `HTTP 500 Internal Server Error`. Perhaps we should catch that and return the same HTTP 400 response?
```suggestion
  try {
    const vaultUrl = new URL(url)
    if (vaultUrl.hostname !== VAULT_DOMAIN) {
      // OR url has expired, check if url has expired using X-Amz-Expires header
      return res.status(400).json({ message: 'This is not a valid Vault url' })
    }
  } catch () {
    return res.status(400).json({ message: 'This is not a valid Vault URL' })
  }
  next()
```

Reply:
good catch! I have added a try catch block now